vault/logical_system.go

```diff
@@ -1025,6 +1045,10 @@ func (b *SystemBackend) handleRemount(ctx context.Context, req *logical.Request,
 			logical.ErrInvalidRequest
 	}
 
+	if err = validateMountPath(toPath); err != nil {
+		return handleError(fmt.Errorf("'to' %v", err))
+	}
```

Comment:
Note that I did not validate the `fromPath` because if we validate it then any Vault users would not be able to move their mounts back to the original name. Also, because the `fromPath` is the path that was already cleaned by `http.ServeMux` and the non-printable characters handler, `fromPath` should only ever be a valid path when a mount is created.